src/endpoints/asset-endpoints.js

```diff
@@ -0,0 +1,93 @@
+const HTTPClient = require('../http-client')
+
+const bulkHeaders = {
+  'Content-Type': 'application/vnd.api+json; ext=bulk',
+  'Accept': 'application/vnd.api+json; ext=bulk'
+}
+
+/**
+ * Bulk add assets
+ *
+ * @param [{ name, reference, remote_file_url }] assets
+ */
+function createAssetsFilesV1 (assets) {
+  const data = assets.map(asset => {
+    return {
+      type: 'asset_files',
+      attributes: {
+        name: asset.name,
+        reference: asset.reference,
+        file_remote_url: asset.remote_file_url
+      },
+      relationships: {
+        asset_folder: {
+          type: 'asset_folders',
+          attributes: {
+            name: 'Product Imagery',
+            reference: 'product_imagery'
+          }
+        }
+      }
+    }
+  })
+
+  const payload = {
+    data: data,
+    meta: {
+      permissions: {
+        records: {
+          asset_files: [ 'create', 'update' ],
+          asset_folders: [ 'create', 'update' ]
+        }
+      }
+    }
+  }
+
+  return HTTPClient.post('v1/asset_files', payload, bulkHeaders).then(response => {
+    return {
+      status: response.status,
+      data: response.data
+    }
+  })
+}
+
+/**
+ * Bulk add mapping between asset files and products
+ *
+ * @param [{ product_reference, asset_file_reference }] req
+ */
+function createProductAssetFilesV1 (mappings) {
+  const data = mappings.map(mapping => {
+    return {
+      type: 'product_asset_files',
+      attributes: {
+        position: 0,
```

Comment:
Again, you probably want this to be dynamic, if a product has 5 images, you'll want them in position 1-5, not all set to 0.

Reply:
What do you mean by 'dynamic'? I can index them in this request easily enough but if I omit the position it will it auto-position them? Otherwise I've got all the fun-and-games of doing a look-up to see how many assets a product already has (and race conditions if lots of images are uploaded at once)

Reply:
I mean allow a value to be passed to this method, e.g. `asset.position`, rather than hard-coding as `0`.

We will require Trendy Golf to use filenames containing a position, e.g. `ABCDEF-1.jpg`, `ABCDEF-2.jpg` etc etc. Then you can just read out the SKU and position using a regex.